utils/production-automation/run-amplicon.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env /hps/nobackup2/production/metagenomics/pipeline/tools-v5/miniconda3-4.6.14/bin/python3
+import logging
+import os
+import re
+import yaml
+import argparse
+from subprocess import CalledProcessError
+from subprocess import check_output
+
+cwlSingle = '/hps/nobackup2/production/metagenomics/pipeline/tools-v5/pipeline-v5/workflows/amplicon-wf-single-empty.cwl'
+cwlPaired = '/hps/nobackup2/production/metagenomics/pipeline/tools-v5/pipeline-v5/workflows/amplicon-wf-paired-empty.cwl'
+ymlSingle = '/hps/nobackup2/production/metagenomics/pipeline/tools-v5/pipeline-v5/workflows/amplicon-wf-single-job.yml'
+ymlPaired = '/hps/nobackup2/production/metagenomics/pipeline/tools-v5/pipeline-v5/workflows/amplicon-wf-paired-job.yml'
+toil_mem= '30G'
+amplicon_mem= 20000
+stats_mem = 1000
+restart_mem= 10000
+toil_cores = 4.0
+lsf_cores = 4
+groupname = '/toil-test'
+queue = 'production-rh74'
+
+def setup_dir(projectID, pathtoProject):
+    runDict = {}
+    readPath = os.path.join(pathtoProject, 'raw')
+    for file in os.listdir(readPath):
+        runName = re.findall('(\w+).fastq', file)[0]
+        runNameStrip = runName.split("_")[0]
+        if not runNameStrip in runDict:
+            if '_' in runName:
+                runDict[runNameStrip] = 'paired'
+            else:
+                runDict[runNameStrip] = 'single'
+    workDir = pathtoProject
+    outDir = os.path.join(pathtoProject, projectID + '-v5-annotation')
+    os.mkdir(outDir)
+    return [runDict, workDir, outDir]
```

Comment:
Did you know that a method could return multiple objects/values? Instead of returning a list object you could return them individually like that:
return runDict, workDir, outDir

Reply:
I think you know it already, I have seen it now further down in your code.